tests/compiler/std/typedarray.ts

```diff
@@ -845,5 +845,45 @@ testTypedArraySet<Float64Array>();
   valuesEqual<Uint8ClampedArray>(targetClampedArray, [1, 255, 100, 255, 0, 0, 100, 10, 255, 0]);
 }
 
+function testArraySort<ArrayType extends TypedArray<T>, T extends number>(): void {
+  var source: ArrayType = instantiate<ArrayType>(3);
+  source[2] = <T>1;
+  source[1] = <T>2;
+  source[0] = <T>3;
+
+  let result = source.sort();
+  assert(result[0] == <T>1);
+  assert(result[1] == <T>2);
+  assert(result[2] == <T>3);
+
+  result = result.sort(function(a: T, b: T) {
```

Comment:
Could've done `b - a` but I didn't want to cast depending on the type of `T`.

Reply:
You could just use 
```ts
result.sort();
```
instead:
```ts
  result = result.sort(function(a: T, b: T) { ... });
```